File: coin_java/src/main/java/cn/stylefeng/guns/modular/mining/entity/MiningOrder.java
package cn.stylefeng.guns.modular.mining.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.io.Serializable;
import java.math.BigDecimal;
import java.util.List;

import cn.stylefeng.guns.modular.base.entity.BaseEntity;
import lombok.experimental.Accessors;

/**
  * <p> 矿机订单 实体类 </p>
  *
  * @author yaying.liu
  * @since 2022-06-08 21:02:12
 */

@Data
@TableName("mining_order")
@Accessors(chain = true)
public class MiningOrder extends BaseEntity implements Serializable {

     private static final long serialVersionUID = 1L;

     /**
      * 矿机订单
      */
     @TableId(value="mining_order_id", type= IdType.AUTO)
     private Long miningOrderId;

     /**
      * 矿机图片
      */
     @TableField("image")
     private String image;


     /**
      * 矿机图片
      */
     @TableField("image2")
     private String image2;

     /**
      * 矿机标题
      */
     @TableField("title")
     private String title;

     /**
      * 活动人
      */
     @TableField("member_id")
     private Long memberId;

     /**
      * 矿机
      */
     @TableField("mining_id")
     private Long miningId;

     /**
      * 矿机名称
      */
     @TableField("name")
     private String name;

     /**
      * 周期量
      */
     @TableField("cycle_number")
     private Integer cycleNumber;

     /**
      * 周期类型 1001年 月1002 日1003
      */
     @TableField("cycle_type")
     private String cycleType;

     /**
      * 挖矿状态（0：待启动，1：挖矿中，2：已结束,3,带启动，）3是启动过的
      */
     @TableField("mining_status")
     private String miningStatus;

     /**
      * 产能币种
      */
     @TableField("mining_coin")
     private String miningCoin;

     /**
      * 静态收益币种
      */
     @TableField("static_coin")
     private String staticCoin;

     /**
      * 预计产能 150 %
      */
     @TableField("estimated_capacity")
     private java.math.BigDecimal estimatedCapacity;

     /**
      * 预计静态化收益率
      */
     @TableField("estimated_static_yield")
     private java.math.BigDecimal estimatedStaticYield;

     /**
      * 燃料能源币种
      */
     @TableField("fuel_energy_coin")
     private String fuelEnergyCoin;

     /**
      * 燃料能源数量
      */
     @TableField("fuel_energy_number")
     private java.math.BigDecimal fuelEnergyNumber;

     /**
      * 能源损耗币种
      */
     @TableField("energy_loss_coin")
     private String energyLossCoin;

     /**
      * 能源损耗数量
      */
     @TableField("energy_loss_number")
     private java.math.BigDecimal energyLossNumber;


     /**
      * 算力币种
      */
     @TableField("calculating_power_coin")
     private String calculatingPowerCoin;

     /**
      * 算力数量
      */
     @TableField("calculating_power_number")
     private java.math.BigDecimal calculatingPowerNumber;


     /**
      * 合约价格币种
      */
     @TableField("price_coin")
     private String priceCoin;

     /**
      * 合约价格数量
      */
     @TableField("price")
     private java.math.BigDecimal price;

     /**
      * 持续天数 持续天数+已挖矿天数 >= 周期量
      */
     @TableField("mining_days")
     private Integer miningDays;

     /**
      * 已挖矿天数
      */
     @TableField("mininged_days")
     private Integer miningedDays;

     /**
      * 原始日产出
      */
     @TableField("mining_daysprofit")
     private java.math.BigDecimal miningDaysprofit;

     /**
      * 当前日产出
      */
     @TableField("current_daysprofit")
     private java.math.BigDecimal currentDaysprofit;

     /**
      * 总产出 已领
      */
     @TableField("total_profit")
     private java.math.BigDecimal totalProfit;

     /**
      * 产能收益
      */
     @TableField("received")
     private java.math.BigDecimal received;

     /**
      * 静态化收益
      */
     @TableField("available")
     private java.math.BigDecimal available;

     //带扣能源
     @TableField("energy_to_be_consumed")
     private java.math.BigDecimal energyToBeConsumed;

     /**
      * 邀请好友（且购买矿机/锁仓）产能增加百分比（为0则不增加）
      */
     @TableField("mining_invite")
     private java.math.BigDecimal miningInvite;

     /**
      * 产能增加上限百分比（为0则无上限）
      */
     @TableField("mining_invitelimit")
     private java.math.BigDecimal miningInvitelimit;

     /**
      * 结束时间
      */
     @TableField("end_time")
     private String endTime;

     /**
      * 删除标志
      */
     @TableField("del")
     private String del;

     /**
      * 备注
      */
     @TableField("remark")
     private String remark;

     /**
      * 只用于前端页面 不用于任何逻辑 0609
      * 表示当前属性不是数据库的字段，但在项目中必须使用
      */
     @TableField(exist = false)
     private List<MiningOrderDetail> miningOrderDetail;




 }

File: coin_agent_java/src/main/webapp/assets/modular/mining/miningOrder/miningOrder.js
layui.use(['table', 'admin', 'ax', 'ztree','laydate'], function () {
    var $ = layui.$;
    var table = layui.table;
    var $ax = layui.ax;
    var admin = layui.admin;
    var $ZTree = layui.ztree;
    var laydate = layui.laydate;

    /**
     * 矿机订单管理初始化
     */
    var Obj = {
        tableId: "miningOrderTable",	//表格id
        condition: {
            miningOrderId: ""
            ,timeLimit:""
        }
    };
    var cycleTypeMap = {
        "1001":"年",
        "1002":"月",
        "1003":"日",
    }
    var statusMap = {
        "0":"待启动",
        "1":"挖矿中",
        "2":"已结束",
        "3":"待启动",
    }
    //渲染时间选择框
    laydate.render({
        elem: '#timeLimit',
        range: true,
        max: Feng.currentDate()
     });
    /**
     * 初始化表格的列
     */
    Obj.initColumn = function () {
        return [[
            // {type: 'checkbox'},
             {field: 'miningOrderId', hide: true, sort: true, title: 'id'},
             {field: 'image', sort: true, templet: "#imgs",title: '矿机图片'},
             {field: 'title', sort: true, title: '矿机标题'},
            {field: 'image2', sort: true,templet: "#imgs2", title: '币种logo'},
             {field: 'memberIdValue', sort: true, title: '活动人'},
             // {field: 'miningId', sort: true, title: '矿机'},
             {field: 'name', sort: true, title: '矿机名称'},

            {field: 'cycleType', sort: true, title: '周期'
                , templet:function (data) {
                    return data.cycleNumber+cycleTypeMap[data.cycleType] ;
                }},
            {field: 'miningStatus', sort: true, title: '挖矿状态'
                , templet:function (data) {
                    return statusMap[data.miningStatus];
                }},
            {field: 'estimatedCapacity', sort: true, title: '目标产能',templet:function (data) {
                    return data.estimatedCapacity+data.miningCoin ;
                }},
            {field: 'estimatedStaticYield', sort: true, title: '静态化目标收益',templet:function (data) {
                    return data.estimatedStaticYield+data.staticCoin ;
                }},
            {field: 'fuelEnergyCoin', sort: true, title: '燃料能源', templet:function (data) {
                    return data.fuelEnergyNumber+data.fuelEnergyCoin ;
                }},
            {field: 'energyLossCoin', sort: true, title: '能源损耗', templet:function (data) {
                    return data.energyLossNumber+data.energyLossCoin ;
                }},
            {field: 'priceCoin', sort: true, title: '合约价格', templet:function (data) {
                    return data.price+data.priceCoin ;
                }},
            {field: 'miningedDays', sort: true, title: '挖矿天数'},
            {field: 'miningDays', sort: true, title: '静态天数'},
             // {field: 'miningDaysprofit', sort: true, title: '原始日产出'},
             // {field: 'currentDaysprofit', sort: true, title: '当前日产出'},
             // {field: 'totalProfit', sort: true, title: '总产出'},
             {field: 'received', sort: true, title: '产能收益'},
             {field: 'available', sort: true, title: '静态化收益'},
             // {field: 'miningInvite', sort: true, title: '邀请好友（且购买矿机/锁仓）产能增加百分比（为0则不增加）'},
             // {field: 'miningInvitelimit', sort: true, title: '产能增加上限百分比（为0则无上限）'},
             // {field: 'endTime', sort: true, title: '结束时间'},
             {field: 'createTime', sort: true, title: '创建时间'},
            // {align: 'center', toolbar: '#tableBar', title: '操作', minWidth: 200}
        ]];
    };


        /**
         * 点击查询按钮
         */
        Obj.search = function () {
            var queryData = {};
            queryData['condition'] = $("#condition").val();
            queryData['miningOrderId'] = Obj.condition.miningOrderId;
            queryData['timeLimit'] = $("#timeLimit").val();
            table.reload(Obj.tableId, {where: queryData});
        };



    /**
     * 打开查看矿机订单详情
     */
    Obj.openMiningOrderDetail = function () {
        if (this.check()) {
            var index = layer.open({
                type: 2,
                maxmin:true,
                title: '矿机订单详情',
                area: ['800px', '420px'], //宽高
                fix: false, //不固定
                maxmin: true,
                content: Feng.ctxPath + '/miningOrder/miningOrder_edit/' + Obj.seItem.id
            });
            this.layerIndex = index;
        }
    };


        /**
         * 删除矿机订单
         *
         */
        Obj.onDeleteMiningOrder = function (data) {
            var operation = function () {
                var ajax = new $ax(Feng.ctxPath + "/miningOrder/delete", function () {
                    Feng.success("删除成功!");
                    table.reload(Obj.tableId);
                }, function (data) {
                    Feng.error("删除失败!" + data.message + "!");
                });
                ajax.set("miningOrderId", data.miningOrderId);
                ajax.start();
            };
            Feng.confirm("是否删除矿机订单 ?", operation);
        };

        /**
         * 弹出添加矿机订单
         */
        Obj.openAddMiningOrder = function () {
            admin.putTempData('formOk', false);
            top.layui.admin.open({
                type: 2,
                maxmin:true,
                title: '添加矿机订单',
                area:["800px","420px"],
                content: Feng.ctxPath + '/miningOrder/miningOrder_add',
                end: function () {
                    admin.getTempData('formOk') && table.reload(Obj.tableId);
                }
            });
        };

        /**
         * 导出excel按钮
         */
        Obj.exportExcel = function () {
            var checkRows = table.checkStatus(Obj.tableId);
            if (checkRows.data.length === 0) {
                Feng.error("请选择要导出的数据");
            } else {
                table.exportFile(tableResult.config.id, checkRows.data, 'xls');
            }
        };
       /**
         * 点击编辑矿机订单
         *
         */
         Obj.onEditMiningOrder = function (data) {
            admin.putTempData('formOk', false);
            top.layui.admin.open({
                type: 2,
                maxmin:true,
                title: '修改矿机订单',
                area:["800px","420px"],
                content: Feng.ctxPath + '/miningOrder/miningOrder_edit?miningOrderId=' + data.miningOrderId,
                end: function () {
                    admin.getTempData('formOk') && table.reload(Obj.tableId);
                }
            });
        };
     // 渲染表格
        var tableResult = table.render({
            elem: '#' + Obj.tableId,
            url: Feng.ctxPath + '/miningOrder/list',
            page: true,
            height: "full-158",
            cellMinWidth: 100,
            //limit:100,
            //limits: [100,200,300,400,500],
            cols: Obj.initColumn()
        });
     // 搜索按钮点击事件
        $('#btnSearch').click(function () {
            Obj.search();
        });

        // 添加按钮点击事件
        $('#btnAdd').click(function () {
            Obj.openAddMiningOrder();
        });

        // 导出excel
        $('#btnExp').click(function () {
            Obj.exportExcel();
        });

        // 工具条点击事件
        table.on('tool(' + Obj.tableId + ')', function (obj) {
            var data = obj.data;
            var layEvent = obj.event;
            if (layEvent === 'edit') {
                Obj.onEditMiningOrder(data);
            } else if (layEvent === 'delete') {
                Obj.onDeleteMiningOrder(data);
            }
        });
});
